ext/sockets/sockets.c

```diff
@@ -1883,11 +1889,29 @@ PHP_FUNCTION(socket_set_option)
 				RETURN_THROWS();
 			}
 
-			convert_to_long(l_onoff);
-			convert_to_long(l_linger);
+			zend_long zl_onoff = zval_try_get_long(l_onoff, &failed);
```

Comment:
Might also use `zval_get_long()` for better BC, but I don't have a strong opinion on this.